crates/biome_js_formatter/src/js/expressions/arrow_function_expression.rs

```diff
@@ -539,30 +539,40 @@ impl Format<JsFormatContext> for ArrowChain {
                     // rest of the arrows in the chain need to format their
                     // comments manually, since they won't have their own
                     // Format node to handle it.
-                    if !is_first_in_chain
-                        && f.context().comments().has_leading_comments(arrow.syntax())
-                    {
-                        // A grouped layout implies that the arrow chain is trying to be rendered
-                        // in a condensend, single-line format (at least the signatures, not
-                        // necessarily the body). In that case, we _need_ to prevent the leading
-                        // comments from inserting line breaks. But if it's _not_ a grouped layout,
-                        // then we want to _force_ the line break so that the leading comments
-                        // don't inadvertently end up on the previous line after the fat arrow.
-                        if is_grouped_call_arg_layout {
-                            write!(f, [space(), format_leading_comments(arrow.syntax())])?;
-                        } else {
-                            write!(
-                                f,
-                                [
-                                    soft_line_break_or_space(),
-                                    format_leading_comments(arrow.syntax())
-                                ]
-                            )?;
+                    let should_format_comments = !is_first_in_chain
+                        && f.context().comments().has_leading_comments(arrow.syntax());
+                    let is_first = is_first_in_chain;
+
+                    let formatted_signature = format_with(|f: &mut JsFormatter| {
```

Comment:
This whole diff is a little bit weird, but this just moves the leading comment inside of the signature formatting, ensuring that it picks up the potential indention that's applied below (line +590).

It fixes this:
```typescript
foo(
  () =>
  // a comment
    () => {}
)
```
which should actually be formatted to
```typescript
foo(
  () =>
    // a comment
    () => {}
)
```
where the leading comment lines up with the following parens.